File: test/Z.IO.Test/System.IO.DirectoryInfo/DirectoryInfo.PathCombine.cs
// Copyright (c) 2015 ZZZ Projects. All rights reserved
// Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
// Website: http://www.zzzprojects.com/
// Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
// All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library

using System;
using System.IO;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Z.IO.Test
{
    [TestClass]
    public class System_IO_DirectoryInfo_PathCombine
    {
        [TestMethod]
        public void PathCombine()
        {
            // Type
            var @this = new DirectoryInfo(AppDomain.CurrentDomain.BaseDirectory);

            string path1 = "Fizz";
            string path2 = "Buzz";

            // Exemples
            string result = @this.PathCombine(path1, path2); // Combine path1 and path2 with the DirectoryInfo

            // Unit Test
            Assert.AreEqual(Path.Combine(@this.FullName, path1, path2), result);
        }
    }
}

File: src (VB.NET)/Z.ExtensionMethods.VB/Z.IO/System.IO.DirectoryInfo/DirectoryInfo.PathCombine.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library

Imports System.Collections.Generic
Imports System.IO
Imports System.Linq

Public Module Extensions_772

	''' <summary>
	'''     Combines multiples string into a path.
	''' </summary>
	''' <param name="this">The @this to act on.</param>
	''' <param name="paths">A variable-length parameters list containing paths.</param>
	''' <returns>
	'''     The combined paths. If one of the specified paths is a zero-length string, this method returns the other path.
	''' </returns>
	<System.Runtime.CompilerServices.Extension> _
	Public Function PathCombine(this As DirectoryInfo, ParamArray paths As String()) As String
		Dim list As List(Of String) = paths.ToList()
		list.Insert(0, this.FullName)
		Return Path.Combine(list.ToArray())
	End Function
End Module


